pandas/core/reshape/merge.py

```diff
@@ -1683,7 +1681,7 @@ def get_join_indexers(
     left_keys: list[ArrayLike],
     right_keys: list[ArrayLike],
     sort: bool = False,
-    how: JoinHow = "inner",
+    how: JoinHow + Literal["leftsemi"] = "inner",
```

Comment:
Is there any reason why we don't just add `leftsemi` to the `JoinHow` definition?